src/ui/router/Browse.js

```diff
@@ -9,22 +9,31 @@ define([
         let browseObject;
         let unobserve = undefined;
         let currentObjectPath;
+        let routingInProgress;
 
         openmct.router.route(/^\/browse\/?$/, navigateToFirstChildOfRoot);
 
         openmct.router.route(/^\/browse\/(.*)$/, (path, results, params) => {
+            routingInProgress = true;
             let navigatePath = results[1];
             navigateToPath(navigatePath, params.view);
+            navigateToObject(null, null, params);
         });
 
-        openmct.router.on('change:params', function (newParams, oldParams, changed) {
+        openmct.router.on('change:params', navigateToObject);
+
+        function navigateToObject(newParams, oldParams, changed) {
```

Comment:
I don't think these changes are needed any more ?

Reply:
Tested: Still need this change for proper routing.

Reply:
updated function name, as per discussion.